src/lib/Microsoft.Health.Events/EventHubProcessor/EventProcessor.cs

```diff
@@ -111,5 +113,18 @@ async Task ProcessInitializingHandler(PartitionInitializingEventArgs initArgs)
                 processor.PartitionInitializingAsync -= ProcessInitializingHandler;
             }
         }
+
+        private (Exception exception, string reason) ProcessErrorEventException(Exception exception)
+        {
+            switch (exception)
+            {
+                case EventHubsException eventHubsException:
```

Comment:
This will cover authorization and invalid host exceptions?

Reply:
The invalid host will be via SocketException and invalid event hub is through EventHubException.
I'm double checking on the auth one.

Reply:
Checked that we get UnauthorizedAccessException separately for authorization errors so handled that explicitly. In general though, given we don't know what the underlying types of exceptions we can expect, unfortunately it could be that most of it fall under "GeneralError" category.